src/_guides/libraries/objective-c-interop.md

```diff
@@ -293,6 +293,58 @@ then check the status, and wait for the duration of the audio file:
     }
 ```
 
+### Callbacks and multithreading limitations
+
+Multithreading issues are the biggest limitation
+of Dart's experimental support for Objective-C interop.
+While `ffigen` does support converting
+Dart functions to Objective-C blocks,
+these limitations must be kept in mind when using them.
+These issues also affect the use of certain Apple APIs.
+
+1. Dart isolates are not the same thing as threads.
+   An isolate is not guaranteed to run on a particular thread,
+   and the VM may change which thread an isolate is running on
+   without warning.
+   There is an [open feature request][] to allow isolates to be
+   pinned to specific theads.
+2. Most Apple APIs don't have any guarantees about
+   which thread a callback will be run on.
+3. Most APIs that involve UI interaction
+   can only be called on the main thread,
+   also known in Flutter as the platform thread.
+4. Many Apple APIs are [not thread safe][].
+
+Points 1 and 2 mean that a callback created in one isolate
+may be invoked on a thread running a different isolate,
+or no isolate at all.
+This will cause your app to crash.
+You can work around this limitation by writing some
+Objective-C code that intercepts your callback and
+forwards it over a [`Dart_Port`]({{site.dart-api}}/dart-ffi/NativePort.html)
+to the correct isolate.
+For an example of this,
+see the implementation of [`package:cupertino_http`][].
+
+Point 3 means that directly calling some Apple APIs
+using the generated Dart bindings may be thread unsafe.
+This could crash your app, or do other unpredicable things.
+You can work around this limitation by writing some
+Objective-C code that dispatches your call
+to the main thread.
+For more information, see the [Objective-C dispatch documentaion][].
+
+Regarding point 4,
+although Dart isolates can switch threads,
```

Comment:
I admit it. I'm confused by talking about isolates AND threads. Unless you mean isolates on Dart and threads on iOS.

Reply:
Hmmm. This whole section is about the relationship between threads and isolates. Maybe it needs to be clarified.

Reply:
Mentioned that this is about the relationship between threads and isolates in the intro.